how-to-guides/re-run-failed-training/notebooks/re_run_failed_training.ipynb

```diff
@@ -0,0 +1,540 @@
+   "source": [
+    "## Introduction\n",
+    "When you are executing a model training script that’s being tracked in Neptune and it fails in the middle, you can easily re-run it with the same metadata such as hyperparameters, data, and code version.\n",
+    "\n",
+    "By the end of this guide, you will learn how to re-open a failed Neptune run to fetch the metadata needed to re-run it and log all metadata from the model training, validation, or testing to a new run, to save results you didn’t get from the failed run.\n",
+    "\n",
+    "[See this example in Neptune](https://app.neptune.ai/common/showroom/e/SHOW-3695)\n",
+    "\n",
+    "[![image](https://files.gitbook.com/v0/b/gitbook-x-prod.appspot.com/o/spaces%2F-MT0sYKbymfLAAtTq4-t%2Fuploads%2Fcgj0RmMm89ejwZKSZfXK%2Fimage.png?alt=media&token=c0314a39-5e8d-465c-9752-a688bd207b91)](https://app.neptune.ai/common/showroom/e/SHOW-3695)\n",
+    "<center><small>Training charts from the run created with the metadata taken from the failed run.</small></center>\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Before you start\n",
+    "\n",
+    "Make sure that you have:\n",
+    "* [Python 3.7+ installed](https://www.python.org/downloads/),\n",
+    "* [Basic familiarity with Neptune (create run and log metadata to it)](hhttps://docs.neptune.ai/you-should-know/what-can-you-log-and-display)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "54CFqKRANLK5",
+    "outputId": "3612c157-e2e0-4092-9f23-ffb2a8846c4f",
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "! pip install -U neptune-client torch==1.10.2 torchvision==0.11.3 #add other dependencies"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "KdpS2pD5HvN2"
+   },
+   "source": [
+    "## Step 1: Get Run ID\n",
+    "You will get the Run ID of the failed run **programmatically**.\n",
+    "\n",
+    "**Note**: To log or retrieve metadata from Neptune, you need the `project name` and the `api_token`.\n",
+    "\n",
+    "To make this example easy to follow, we have created a public project **'common/showroom'** and a shared user **'neptuner'** with the API token **'ANONYMOUS'**. As you will see in the code cell below.\n",
+    "\n",
+    "**(Optional)** If you want to log to your own project you have to have or create [Neptune account](https://app.neptune.ai/register/) and [project](https://docs.neptune.ai/getting-started/installation#setting-the-project-name).\n",
+    "Then you can pass [project](https://docs.neptune.ai/getting-started/installation#setting-the-project-name) and [api_token](https://docs.neptune.ai/getting-started/installation#authentication-neptune-api-token) arguments to the `init()` method.\n",
+    "\n",
+    "`run = neptune.init(api_token='<YOUR_API_TOKEN>', project='<YOUR_WORKSPACE/YOUR_PROJECT>')` \n"
```

Comment:
Format code snippets within comments with black, and follow latest styleguide (no `<`, `>`)